python/cuda_parallel/cuda/parallel/experimental/_iterators.py

```diff
@@ -13,32 +16,9 @@
 
 _DEVICE_POINTER_SIZE = 8
 _DEVICE_POINTER_BITWIDTH = _DEVICE_POINTER_SIZE * 8
-_DISTANCE_NUMBA_TYPE = numba.types.uint64
-_DISTANCE_IR_TYPE = ir.IntType(64)
-_CHAR_PTR_NUMBA_TYPE = numba.types.CPointer(numba.types.int8)
-_CHAR_PTR_IR_TYPE = ir.PointerType(ir.IntType(8))
-
-
-def numba_type_from_any(value_type):
-    return getattr(numba.types, str(value_type))
-
-
-def _sizeof_numba_type(ntype):
-    mapping = {
-        numba.types.int8: 1,
-        numba.types.int16: 2,
-        numba.types.int32: 4,
-        numba.types.int64: 8,
-        numba.types.uint8: 1,
-        numba.types.uint16: 2,
-        numba.types.uint32: 4,
-        numba.types.uint64: 8,
-        numba.types.float32: 4,
-        numba.types.float64: 8,
-    }
-    return mapping[ntype]
 
 
+@lru_cache(maxsize=256)
```

Comment:
The function is just a dictionary lookup anyway.

Maybe simpler is better here?

```
NUMBA_TO_CTYPES_MAPPING = {
    numba.types.int8: ctypes.c_int8,
    numba.types.int16: ctypes.c_int16,
    numba.types.int32: ctypes.c_int32,
    numba.types.int64: ctypes.c_int64,
    numba.types.uint8: ctypes.c_uint8,
    numba.types.uint16: ctypes.c_uint16,
    numba.types.uint32: ctypes.c_uint32,
    numba.types.uint64: ctypes.c_uint64,
    numba.types.float32: ctypes.c_float,
    numba.types.float64: ctypes.c_double,
}
```

Then replace the function calls with `NUMBA_TO_CTYPES_MAPPING[ntype]`.

Reply:
In https://github.com/NVIDIA/cccl/pull/3118/commits/1bffc9878bf65da121c70f859ae60cfbf7f9c9bb I replaced the use of these utility functions with ones that numba provides.

Reply:
Awesome, much better.